src/react/workflow/__tests__/ExpirationForm.test.tsx

```diff
@@ -0,0 +1,111 @@
+const accountName = 'pat';
+const expirationId = 'expirationId';
+
+const S3BucketList = [
+  {
+    bucketName: 'bucket1',
+    bucketRegion: 'us-east-1',
+    bucketArn: 'arn:aws:s3:::bucket1',
+    bucketUrl: 'https://s3.amazonaws.com/bucket1',
+  },
+  {
+    bucketName: 'bucket2',
+    bucketRegion: 'us-east-1',
+    bucketArn: 'arn:aws:s3:::bucket2',
+    bucketUrl: 'https://s3.amazonaws.com/bucket2',
+  },
+];
+
+const locations = [
+  {
+    name: 'locationName1',
+    locationType: 'location-file-v1',
+    details: 'locationDetails1',
+    objectId: 'locationId1',
+    isTransient: false,
+    isBuiltin: false,
+    sizeLimitGB: false,
+    legacyAwsBehavior: false,
+  },
+  ]
+
+const server = setupServer(rest.post(
+  `${TEST_API_BASE_URL}/api/v1/instance/${instanceId}/accounts/${accountName}/workflows/${expirationId}`,
+  (req, res, ctx) => res(ctx.json([])),
+),);
+
+jest.mock("react-hook-form", () => ({
+  ...jest.requireActual("react-hook-form"),
+  useFormContext: () => ( {
+    register: jest.fn(),
+    unregister: jest.fn(),
+    watch: jest.fn(),
+    getValues : jest.fn(),
+    errors: {},
+    formState: {
+      isSubmitting: false,
+      isValid: true,
+    },
+  }),
+}));
+
+beforeAll(() => {
+  server.listen({ onUnhandledRequest: 'error' });
+  mockOffsetSize(200, 800);
+  jest.setTimeout(10_000);
+});
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+const WithFormProvider = ({children}) => {
+      const formMethods = useForm();
+      return <FormProvider {...formMethods}>{children}</FormProvider>
+}
+
+describe('ExpirationForm', () => {
+  it('should render a form for expiration workflow', async () => {
+    try {
+      render(<WithFormProvider><ExpirationForm bucketList={S3BucketList} locations={locations}/></WithFormProvider>, {
+        wrapper
+      });
+
+      await waitFor(() => screen.getByText(/General/i));
+
+      expect(screen.getByText('State')).toBeInTheDocument();
+      expect(screen.getByText('Source')).toBeInTheDocument();
+      expect(screen.getByText('Bucket Name')).toBeInTheDocument();
+      expect(screen.getByText('Filters (optional)')).toBeInTheDocument();
+      expect(screen.getByText('Prefix')).toBeInTheDocument();
+      expect(screen.getByText('Action')).toBeInTheDocument();
+      expect(screen.getByText('Current')).toBeInTheDocument();
+      expect(screen.getByText('Previous')).toBeInTheDocument();
+      expect(screen.getByText('Remove expired Delete markers')).toBeInTheDocument();
+      expect(screen.getByText('Expire incomplete Multipart uploads')).toBeInTheDocument();
+
+      const spinButton = screen.getAllByRole('spinbutton');
+      expect(spinButton[0].getAttribute('type')).toBe('number');
+      expect(spinButton[1].getAttribute('type')).toBe('number');
+      expect(spinButton[2].getAttribute('type')).toBe('number');
+
+    } catch (e) {
+      console.log('should render a form for expiration workflow: ', e);
+      throw e;
+    }
+  });
```

Comment:
This test is not enough we want to toggle Current version and ensure delete marker toggle get disabled, also we want to check that the form is valid only when all required filled are correctly filled in